sphinx/themes/basic/static/searchtools.js

```diff
@@ -514,8 +514,8 @@ const Search = {
     searchTerms.forEach((word) => {
       const files = [];
       const arr = [
-        { files: terms[word], score: Scorer.term },
-        { files: titleTerms[word], score: Scorer.title },
+        { files: terms.hasOwnProperty(word) ? terms[word] : [], score: Scorer.term },
+        { files: titleTerms.hasOwnProperty(word) ? titleTerms[word] : [], score: Scorer.title },
```

Comment:
Preparing that comment has made me realize: we perform an `===` equality check against the `undefined` value further down the function.  We should probably not alter that behaviour in this PR.